### ISSUE TITLE
Question about conditioning mismatch in Oracle-Guided Bidirectional Distillation

### ISSUE THREAD

**Issue by lxxie298** [Soul-AILab/SoulX-FlashHead#77]
Hi authors,

Thank you for releasing SoulX-FlashHead and sharing the technical report 🎉🎉🎉. I have a question about the conditioning mismatch in the Oracle-Guided Bidirectional Distillation stage. 

According to Section 3.2:

- The student generator performs autoregressive rollout using its own generated motion history, `m_pred`.
- The real-score teacher uses the corresponding ground-truth motion history, `m_gt`.
- The fake-score network tracks the student distribution generated under `m_pred`.

Therefore, my understanding of the DMD gradient is approximately:

```
grad_theta(L_DMD)  ∝  -E[ ( s_real(x_t | m_gt) − s_fake(x_t | m_pred) ) · dG_theta/dtheta ]
```

Here, both score networks evaluate the same noised student sample `x_t`, but under different motion-history conditions.

I understand the motivation: the ground-truth-conditioned teacher acts as a privileged oracle that guides a drifting student trajectory back toward the clean data manifold. However, I am unsure whether this target remains realizable when the student history has already deviated significantly from the ground-truth history.

For example, if `m_pred` differs substantially from `m_gt` in pose, motion phase, or facial state, the ground-truth continuation may no longer be a temporally natural continuation of the student's current state. In that case:

1. The real and fake score functions correspond to different conditional distributions:

```
p_phi(x_0 | m_gt)    vs.    p_theta(x_0 | m_pred)
```

2. The DMD gradient may pull the student toward a sample that is likely under the ground-truth history, but not necessarily continuous with the student's generated history.

3. The latent regression loss

```
L_reg = || z_student^(k) − z_gt^(k) ||_2^2
```

further enforces alignment with the ground-truth trajectory, which may conflict with temporal continuity from `m_pred`, especially when the student has already drifted far from `m_gt`.

Could you please clarify the following points?

**1.** How do you ensure that the ground-truth continuation remains reachable from the student's current generated history, especially when `m_pred` has significantly diverged from `m_gt`?

**2.** Did you observe any gradient conflict between `L_DMD` and `L_reg`, especially for later rollout chunks? For example, have you measured the cosine similarity between their gradients as a function of rollout depth or the distance between `m_pred` and `m_gt`?

**3.** Have you compared Oracle conditioning against a same-condition baseline where the real-score teacher is also conditioned on `m_pred`?

| Variant | Teacher condition | Student / fake condition |
|---|---|---|
| Oracle (proposed) | `m_gt` | `m_pred` |
| Same-condition baseline | `m_pred` | `m_pred` |

I may have misunderstood some implementation details because the training code is not currently available. Any clarification would be very helpful.

Thank you again for the excellent work. ❤️❤️❤️
